libs/sdk-backend-spi/src/user/index.ts

```diff
@@ -1,39 +1,64 @@
-// (C) 2020 GoodData Corporation
+// (C) 2020-2021 GoodData Corporation
 import { ObjRef } from "@gooddata/sdk-model";
 import { IUserSettingsService } from "./settings";
+import invariant from "ts-invariant";
 
 /**
- * Represents platform user
+ * Represents platform user.
  *
  * @alpha
  */
 export interface IUser {
     /**
-     * Stored user reference
+     * Stored user reference.
      */
     ref: ObjRef;
 
     /**
-     * Login - unique user ID for logging into the platform
+     * Login - unique user ID for logging into the platform.
      */
     login: string;
 
     /**
-     * Contact email of the user
+     * Contact email of the user.
      */
     email?: string;
 
     /**
-     * First name
+     * Full name.
+     *
+     * Note: This property has higher priority than firstName / lastName.
+     * As users can be stored differently across different backend implementations
+     * (some of them supports only fullName, and some of them only combination of firstName and lastName),
+     * we are putting here a contract, that every spi backend implementation have to fill this property,
+     * if user name is available.
+     */
+    fullName?: string;
+
+    /**
+     * First name - when backend implementations supports it.
      */
     firstName?: string;
 
     /**
-     * Last name
+     * Last name - when backend implementations supports it.
      */
     lastName?: string;
 }
 
+/**
+ * Gets the user full name
+ *
+ * @param user - user to get full name of
+ * @returns the user full name
+ * @alpha
+ */
+export function getUserFullName(user: IUser): string | undefined {
```

Comment:
Naming: we do not use the `get` prefix in other similar functions (like `insightTitle`). Should we use just `userFullName` here?

Reply:
🤦 oh right